gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/param/AnalyticsParam.java

```diff
@@ -135,32 +126,37 @@ public void setSize(int size) {
         this.size = size;
     }
 
-    public AnalyticsTypeParam getTypeParam() {
+    public AnalyticsType getType() {
         return type;
     }
 
-    public void setTypeParam(AnalyticsTypeParam type) {
+    public void setType(AnalyticsType type) {
         this.type = type;
     }
 
-    public AnalyticsTypeParam.AnalyticsType getType() {
-        return type.getValue();
-    }
-
     public List<Range> getRanges() {
-        return (ranges == null) ? null : ranges.getValue();
+        return (ranges == null) ? null : ranges.getValues();
     }
 
     public List<Aggregation> getAggregations() {
-        return (aggs == null) ? null : aggs.getValue();
+        return (aggregations == null) ? null : aggregations.getValues();
+    }
+
+    @QueryParam("order")
+    @Parameter(description = "The field used to sort results. Can be asc or desc (prefix with minus '-') ",
+            example = "order:-response-time")
+    public void setOrder(String param) {
+        if (param != null){
+            orderParam = Order.parse(param);
+        }
     }
 
-    public OrderParam.Order getOrder() {
-        return (order == null) ? null : order.getValue();
+    public Order getOrder() {
+        return orderParam;
     }
 
     public void validate() throws WebApplicationException {
-        if (type.getValue() == null) {
+        if (type == null) {
```

Comment:
Since type is a enum value, it can't be null here. The parsing of the query param will result in a `IllegalArgumentException` if the value does not exist.
I've found this workaround that seems to do the job. WDYT ?
https://stackoverflow.com/a/36216666

Or maybe this can be achieve in an other way.

Reply:
> To my mind, the modification of the QueryParam are not "safe" because, with "xxxParam" classes we are able to manage wrong values of enum.
> 
> Do you think it would be possible to manage this case ?

The right way to handle wrong enum constants is to return HTTP 400.
I would be very strict here, so no conversion from lower to upper case at all.
If an API defines an concrete enum constant, and the consumer of the API is not passing the defined value, we must fail with 400.
Normally spring (MVC) is taking care about this, but as it seems the current code fails with HTTP 500 instead of 400 in case a wrong enum constant is passed.

Reply:
I added the handling for wrong enum values:

https://github.com/gravitee-io/gravitee-management-rest-api/blob/4cea64293876ef09590b64c972d9c459dd7e7b39/gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/provider/ParamExceptionMapper.java#L25-L40